### PR TITLE
[typescript-fetch] Fix code generation for oneOf cases without discriminator

### PR DESCRIPTION
**Pull request by odiak**
## TL;DR

- Fix type error on generated model code using `oneOf` without discriminator.
- Add a sample case to verify this fix.

## Detail

I improved code generation for model with `oneOf` without discriminator on #18702.
But later I found generated code cause type error. (But there is no problem on runtime.)
ref: [comment](https://github.com/OpenAPITools/openapi-generator/pull/18702#issuecomment-2217152688)

The cause of the problem is lack of return statement at the end of functions like `MyModelFromJSONTyped`.
So I added `return json` at the end of these functions.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by odiak** [comment]
Dear technical committee members of TypeScript, could you check this? 🙇
@TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx @macjohnny @topce @akehir @petejohansonxo @amakhrov @davidgamero @mkusaka

File: modules/openapi-generator/src/main/resources/typescript-fetch/modelOneOf.mustache [hunk 1]
```diff
@@ -36,6 +36,8 @@ export function {{classname}}FromJSONTyped(json: any, ignoreDiscriminator: boole
         return {{{.}}}FromJSONTyped(json, true);
     }
     {{/oneOf}}
+
+    return json;
```
Comment (amakhrov) [inline modules/openapi-generator/src/main/resources/typescript-fetch/modelOneOf.mustache:40]: **thought (non-blocking):** 
Here, the fallback case returns the raw `json`.  But a similar fallback in the `xxxToJSON` (below) returns `{}`.

I don't think it makes a huge practical difference (if this happens for real, one should go and fix the code to be compatible with the spec, instead of finding smart ways around it).

Just looks a bit inconsistent.

**Review by amakhrov — APPROVED** [review]
Thanks @odiak - looks good to me!

File: modules/openapi-generator/src/main/resources/typescript-fetch/modelOneOf.mustache (hunk 1 above)
Comment (odiak) [reply modules/openapi-generator/src/main/resources/typescript-fetch/modelOneOf.mustache]: Indeed. I change it to `return {} as any`.

(I needed to add type-assertion because `{}` is not compatible with the return type.
But `json` is also `any`. There is no big difference.)
